pages/doc.js

```diff
@@ -94,48 +77,90 @@ export default class Documentation extends Component {
     })
   }
 
-  getLinkHref = (section, file) => {
-    const sectionSlug =
-      sidebar[section].slug || kebabCase(sidebar[section].name)
-    const fileSlug = file ? kebabCase(file.slice(0, -3)) : undefined
-    return `/doc/${compact([sectionSlug, fileSlug]).join('/')}`
+  getLinkHref = (section, subsection = null, file = null) => {
+    let sect = sidebar[section]
+    let removeExtFunc = filename =>
+      SidebarMenuHelper.removeExtensionFromFileName(filename)
+    const sectionSlug = removeExtFunc(sect.indexFile) || kebabCase(sect.name)
+    const subsectionSlug =
+      (subsection && removeExtFunc(sect.files[subsection].indexFile)) ||
+      sect.files[subsection]
+    const fileSlug = removeExtFunc(file) || (file && file.files[0])
```

Comment:
three lines above extremely hard to understand. For example why file is not actually file sometimes? is there a really strong reason here?

The same for subsection.

Reply:
When parsing, the next element in the sidebar.json structure may turn out to be a string, this is because each level may contain a different structure and a different set of fields.